pkg/cache/internal/informers_map.go

```diff
@@ -65,6 +66,7 @@ func newSpecificInformersMap(config *rest.Config,
 		createListWatcher: createListWatcher,
 		namespace:         namespace,
 		selectors:         selectors,
+		disableDeepCopy:   disableDeepCopy,
```

Comment:
Generally this looks good, thank you for doing this change.

Only thing I was wondering is if it would make sense to key this by object type (maybe with a `*` wildcard to make it global). WDYT @vincepri @joelanford ?

Reply:
I'm ok with that. The type has to be defined in `pkg/client` for both `pkg/cache` and `pkg/cache/internal` need to use it.
WDYT?

Reply:
Updated. I agree with that, for we might need to add some optional policies for it in future.

Reply:
@alvaroaleman I also considered the need for this to be GVK-specific. My [comment earlier](https://github.com/kubernetes-sigs/controller-runtime/pull/1274#issuecomment-887974926) touches on that:
> This option would probably require something like the GVKCaches composite cache proposed in #1591, which allows separate cache implementations per GVK.

It's probably do-able to make this option work in the informersMap on a per-GVK basis, but I feel like there are going to be other use cases around GVK-specific cache semantics (e.g. watch GVKA in namespace `foo`, GVKB in namespace `bar` and GVKC cluster-wide) that it may be better to keep this minimal.

That said, I see we already have selectors keyed by GVK, so maybe its best to do the same here and re-evaluate where GVK-specifics should live in a separate feature?

Reply:
> That said, I see we already have selectors keyed by GVK, so maybe its best to do the same here and re-evaluate where GVK-specifics should live in a separate feature?

Yeah, that was what I thought, because otherwise we end up in a situation where ppl have to set different options on different levels and that will be pretty confusing. Discussing what option we want to be configurable by GVK how and where in a separate issue is a good idea :+1: